drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1222,6 +1222,9 @@ static void endrx_isr(const struct device *dev)
 		/* Remove the short until the subsequent next buffer is setup */
 		nrf_uarte_shorts_disable(uarte, NRF_UARTE_SHORT_ENDRX_STARTRX);
 	} else {
+		data->async->rx_buf = NULL;
```

Comment:
I think it would be better to modify `notify_rx_buf_release()` instead, so that it would always clear the passed pointer (it seems that there is no need for this function to have the `clear` parameter).

Reply:
Sounds good. I modified the `notify_rx_buf_release` function. I am not sure about changing function name, every function that trigger event have name started from `notify_`

Reply:
Discussed offline. I changed function name to `release_rx_buf`